lib/player.js

```diff
@@ -2405,7 +2405,7 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     const setupPlayhead = (startTime) => {
       this.playhead_ = this.createPlayhead(startTime);
       this.playheadObservers_ =
-          this.createPlayheadObserversForMSE_(startTimeOfLoad);
+          this.createPlayheadObserversForMSE_(startTime);
```

Comment:
I could use more explanation in the PR description.  Why is startTime correct and startTimeOfLoad wrong?  What's the difference?

Might also be worth a comment here in the code.

Reply:
There are extensive notes in the associated issue #6711

Reply:
We don't understand why `startTimeOfLoad` was *ever* used, especially for VOD. The `startTimeOfLoad` argument is used to set the `startsPastZero` flag in `RegionObserver`. The comments there seem to indicate that it is playhead related:
https://github.com/shaka-project/shaka-player/blob/30ac8c00259395da532cacea7bf41c84510c9e9d/lib/media/region_observer.js#L39-L44
This makes sense. If the user called `player.load()` with a non-zero start time, the observer should wait until the initial seek has completed. `startTimeOfLoad` is the wall clock time of when the load operation started. How does that play into how the region observer operates? Before the commit mentioned in the issue, `startTimeOfLoad` was alway `NaN` at the time `createPlayheadObserversForMSE_`, which coincidentally resulted in the correct behavior (for VOD).

Reply:
@theodab Can you provide any insight here? The comments in #5361 state:

> This change modifiers the region observer so that it does not start examining regions until the time gets past 0 seconds, except in the case of VOD.

Unfortunately, after the @avelad's [state engine refactor](https://github.com/shaka-project/shaka-player/commit/4425dca2836d4820e7d1b7315d52b67b465cc8ec), `startTimeOfLoad` is always greater than zero, which means `startsPastZero` is always `true`, regardless of live/vod.

Reply:
Yeah, I think the original code using `startTimeOfLoad` was a mistake. It's supposed to check for if we were loading partway through the presentation, so that we won't fire region events for regions that the player loads in after the end of. The `startTime` variable would be the actual variable we want for this.

Honestly, we should probably just rename the `startTImeOfLoad` variable. It's confusingly similar to `startTime`, despite the two variables having much different meanings and effects in the loading process.